### PR TITLE
[AzureStack Backup] Add session records and enable TestRestoreBackup

### PR DESCRIPTION
**Pull request by daozha**
## Description
There was a skipped case in AzureStack BackupAdmin, enable that case and add session records for that

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/AzureStack/Admin/BackupAdmin/Backup.Admin.Tests/src/BackupTests.cs [hunk 1]
```diff
@@ -62,15 +62,15 @@ public class BackupTests : BackupTestBase
...
+        [Fact]
         public void TestRestoreBackup()
         {
             RunTest((client) =>
             {
                 var backupLocation = "local";
                 var backup = client.BackupLocations.CreateBackup(ResourceGroupName, backupLocation);
                 Assert.NotNull(backup);
-                client.Backups.Restore(backupLocation, ResourceGroupName, backup.Name);
+                client.Backups.Restore(backupLocation, ResourceGroupName, backup.BackupId);
```
Comment (deathly809) [inline src/AzureStack/Admin/BackupAdmin/Backup.Admin.Tests/src/BackupTests.cs]: You should use the name of the resource.  We should file a bug against the AutoRest team to handle nested resources correctly.

Comment (daozha) [reply src/AzureStack/Admin/BackupAdmin/Backup.Admin.Tests/src/BackupTests.cs]: Make sense. So you mean we should create an issue in Azure/autorest?

Comment (deathly809) [reply src/AzureStack/Admin/BackupAdmin/Backup.Admin.Tests/src/BackupTests.cs]: Yeah I think so.  I will talk with Bala and the AutoRest team

**Comment by dsgouda** [comment]
Will merge once @deathly809 approves

**Comment by deathly809** [comment]
@dsgouda It looks fine to me.

**Comment by dsgouda** [comment]
@azuresdkci Test this please

**Pull request merged**